Documentation/hubble.rst

```diff
@@ -0,0 +1,101 @@
+****************
+Hubble internals
+****************
+
+.. note:: This documentation section is targeted at developers who are willing
+          to contribute to Hubble. For this purpose, it describes Hubble
+          internals.
+
+.. note:: This documentation covers the Hubble server (sometimes referred as
+          "Hubble embedded") and Hubble-relay components but does not cover the
+          Hubble UI and CLI.
+
+Hubble builds on top of Cilium and eBPF to enable deep visibility into the
+communication and behavior of services as well as the networking infrastructure
+in a completely transparent manner.
+
+One of the design goals of Hubble is to achieve all of this at large scale.
+Although the initial release of Hubble was a standalone product, as of Cilium
```

Comment:
Replace product

Reply:
I replaced product by component (I'm open to suggestions).